articles/linear-regression-r/index.md

```diff
@@ -0,0 +1,306 @@
+Coefficients:
+             Estimate Std. Error t value Pr(>|t|)    
+(Intercept)  3.526667   0.374290   9.422   <2e-16 ***
+youtube      0.045765   0.001395  32.809   <2e-16 ***
+facebook     0.188530   0.008611  21.893   <2e-16 ***
+newspaper   -0.001037   0.005871  -0.177     0.86    
+---
+Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
+
+```
+The column labeled Estimation contains regression coefficients(β's).
+If any value of β is zero then it means that the variable associated with such a  β is not significant to the model. The coefficients we are having in our model are just but estimates and they can never be zero. To know how possibly a coefficient can be zero, we make use of t statistic and p-value which in our summary are labeled t value and pr(<|t|)
+respectively. The p-value is a probability that gauges the likelihood of a coefficient being not significant to the model.  The conventional limit of p-value to determine coefficient significance is 0.005. In our summary, the newspaper has a p-value of 0.86.
+This indicates that the newspaper is not significant and should be removed from our model. For a quick way for identification of which variable is significant to the model, we consider the asterisks(***). The three asterisks in the summary indicate highly significant coefficients to the model. The newsletter is not indicated hence not significant.
+
+(iv) F-statistic
+```r
+F-statistic: 570.3 on 3 and 196 DF,  p-value: < 2.2e-16
+
+```
+This statistic shows the overall significance of our model. Model is only significant if all coefficients are nonzero and it is not significant if all coefficients are zero. As we had already said in the coefficients section, coefficients being estimates they can never be zero. We there rely on p-value to declare whether our model is significant or not. A model is significant if its p-value less or equal to 0.05 otherwise it is insignificant. Our model has a p-value < 2.2e-16 which is much smaller hence our model is significant.
+
+#### step 6. Analyzing the accuracy of our model
+Now that we have our model, let's test how good it is. To do this, we shall use our test set that we had separated in step 3 and make our model predict sales values for those data points. Because we have the actual sales values for the same points, we shall compare them and see how our model performs. We enter the code shown below.
+
+```r
+#Predicting
+numx<-data_size[1]*(1-splitRation)
+x_axis<-seq(numx)
+dataframe<-data.frame(x_axis,Predict, test$sales)
+#Library required
+library(ggplot2)
+Predplot<-ggplot(dataframe,aes(x=x_axis))
+Predplot<-Predplot+geom_line(aes(y=Predict, colour="Predicted"))
+Predplot<-Predplot + geom_point(aes(x=x_axis,y=Predict, colour="Predicted"))
+Predplot<-Predplot + geom_line(aes(y=test$sales, colour="Actual"))
+Predplot<-Predplot + scale_colour_manual("", values = c(Predicted = "red", Actual = "black"))
+Predplot
+
+```
+
+Running the above code we obtain the graph below.
+
```

Comment:
Please explain the plotting part. Beginners will not be familiar with it. Or make sure to add it as a prerequisite.